src/passes/GenerateDynCalls.cpp

```diff
@@ -55,7 +55,8 @@ struct GenerateDynCalls : public WalkerPass<PostWalker<GenerateDynCalls>> {
 
   void visitFunction(Function* func) {
     // Generate dynCalls for invokes
-    if (func->imported() && func->base.startsWith("invoke_")) {
+    if (func->imported() && func->module == ENV &&
+        func->base.startsWith("invoke_")) {
```

Comment:
This seems like maybe unneeded/unrelated?

If you do want to make this change you you should change the logic in src/wasm/wasm-emscripten.cpp which seems do duplicate this when generating invokeFuncs mabye?

Reply:
It was suggested in https://github.com/WebAssembly/binaryen/pull/3191#discussion_r498557152, and yeah, I think the condition in wasm/wasm-emscripten.cpp should be the same. Should I duplicate the condition there? @kripken

Reply:
Added the same condition in wasm-emscripten.cpp.